src/main/java/org/jenkinsci/plugins/assembla/AssemblaBuildTrigger.java

```diff
@@ -245,6 +253,18 @@ public static AssemblaBuildTrigger getTrigger(AbstractProject project) {
         return values;
     }
 
+    private Boolean shouldMergeRequestTriggerBuild(AssemblaMergeRequestCause cause) {
+        if ((cause.isCreated() || cause.isUpdated()) && buildOnMergeRequestEnabled) {
+            return true;
+        } else if (cause.isMerged() && buildOnMergeRequestMergedEnabled) {
```

Comment:
Looks like we don't really need else if, since return would exit the function anyway. So
```
if ((cause.isCreated() || cause.isUpdated()) && buildOnMergeRequestEnabled) {
    return true;
}
if (cause.isMerged() && buildOnMergeRequestMergedEnabled) {
    return true;
}
return cause.isIgnored() && buildOnMergeRequestIgnoredEnabled;
```

Reply:
@speedywizard I adjusted this to remove the "else if", but I left it as three different if-statements because I thought it was easier to understand.

Reply:
@mediabounds , yeah, right, that absolutely makes sense